tests/bindings/test_bindings.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for PennyLane-Lightning Pybind11 bindings."""
+
+import importlib
+import inspect
+
+import pytest
+from conftest import (  # tested device
+    LightningAdjointJacobian,
+    LightningDevice,
+    LightningStateVector,
+    device_name,
+)
+
+
+def get_module_attributes(module_name):
+    """Get classes and functions from a module."""
+    try:
+        module = importlib.import_module(module_name)
+
+        classes = []
+        functions = []
+
+        for name in dir(module):
+            if name.startswith("_") and not name.startswith("__"):
+                continue
+
+            attr = getattr(module, name)
+            if inspect.isclass(attr):
+                classes.append(name)
+            elif inspect.isfunction(attr) or callable(attr):
+                functions.append(name)
+
+        return {"importable": True, "classes": classes, "functions": functions, "module": module}
+    except ImportError as e:
+        return {"importable": False, "error": str(e)}
+
+
+@pytest.mark.parametrize(
+    "module_name",
+    [
+        "pennylane_lightning.lightning_qubit",
+        "pennylane_lightning.lightning_kokkos",
+        "pennylane_lightning.lightning_gpu",
+    ],
+)
+def test_module_importable(module_name):
+    """Test if module can be imported."""
+    result = get_module_attributes(module_name)
+
+    # Skip test if module is not available instead of failing
+    if not result["importable"]:
+        pytest.skip(f"Module {module_name} not available: {result.get('error')}")
+
+    assert result["importable"]
```

Comment:
Do we need this assert statement given this condition? Is the plan to remove the if statement eventually and leave the assert?

Reply:
The plan should be to keep this base PR as simple as possible so we can finalize it asap and prevent any direct commits to it. Adding `NanobindBindings` is a feature that could be included in a separate feature PR (based on this current branch)